File: BuzzeriOS/Buzzer/Model/Device.m
//
//  Device.m
//  Buzzer
//
//  Created by Scot McIntosh on 7/30/14.
//  Copyright (c) 2014 ___SKUNKd___. All rights reserved.
//

#import "Device.h"

@implementation Device

- (id)initDevicewithId:(NSString *)deviceId andName:(NSString *)deviceName
               andInfo:(NSString *)deviceInfo
              andOwner:(NSString *)deviceOwner
             andStream:(NSString *)deviceStream {
    
    self = [super init];
    if (self) {
        _id = deviceId;
        _name = deviceName;
        _info = deviceInfo;
        _owner = deviceOwner;
        _stream = deviceStream;
        _isSignaled = false;        
    }
        
    
    return self;
}

@end

File: Azure/MeetingBuzzer/MeetingBuzzer/Models/Device.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace MeetingBuzzer.Models
{
    public class Device
    {
        public string Id { get; set; }
        public string Name { get; set; }
        public string DeviceInfo { get; set; }
        public string DeviceOwner { get; set; }
        public string Stream { get; set; }
        public bool Signal { get; set; }

        public Device()
        {
            Signal = false;
        }

        public bool isSignaled()
        {
            bool isSignaled = Signal;
            Signal = false;

            return isSignaled;
        }

    }
}